qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -166,11 +167,19 @@ class OptimizerSupportLevel(IntEnum):
     required = 3  # Feature is required and must be given, None is invalid
 
 
+class OptimizerCallback(Protocol):
+    """Callback protocol for optimizer."""
```

Comment:
Can we describe the callback here as to what `xk` means and the bool return so the user can know what to expect of the callback they provide (scipy minimize for instance has a much fuller description of the callback)